### DISCUSSION TITLE
How to implement leaf decay (best practise)

### DISCUSSION THREAD

**Discussion by ElAndree** [Minestom/Minestom#778] [Help] [answered]
Hey there,

got a short question today:
Could somebody give me a hint on how to implement performantly the leaf decay in minestom?
(InstanceTickEvent? Block Handler? idk?)

Thanks!

**Comment by Flamgop** [comment]
I believe leaf decay in vanilla is handled by random tick.

**Reply by ElAndree** [reply]
What would be the equivalent in minestom?

**Comment by MrlnHi** [comment, marked as answer]
In Vanilla it's handled by random ticks, so you just need to steal mojang's random tick system. This is how I believe it works: every tick (InstanceTickEvent) get chunks that are close enough to players to have random ticks. Then from each of those chunk's sections (16 x 16 x 16) choose a set amount (by default 3; game rule randomTickSpeed) of blocks. Then for each of those blocks there's a chance (however idk leaf decay chance) to receive a random tick. However you should probably look into Minecraft's code if you really wanna implement it
